aws-java-sdk-core/src/main/java/com/amazonaws/retry/PredefinedRetryPolicies.java

```diff
@@ -199,6 +198,10 @@ public boolean shouldRetry(AmazonWebServiceRequest originalRequest,
         /** Maximum exponential back-off time before retrying a request */
         private static final int MAX_BACKOFF_IN_MILLISECONDS = 20 * 1000;
 
+        /** Maximum number of retries before the max backoff will be hit */
+        private static final int MAX_RETRIES_BEFORE_MAX_BACKOFF =
```

Comment:
because if someone changes the other constants, then 6 will no longer be correct.  This value actually depends on the other values here.  In my experience, people will make changes to these when they decide there are better values and not realize they need to change this.  Then you could be sad.